src/main/java/ch/njol/skript/expressions/ExprLastDeath.java

```diff
@@ -0,0 +1,137 @@
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.expressions;
+
+import org.bukkit.Statistic;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.Date;
+import ch.njol.skript.util.Timespan;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+
+@Name("Last Death")
+@Description({"The time of the last death of a player.", 
+				"A change to this value can't be negative, so it will be reset or remain unchanged."})
+@Examples("send \"Your last death was %last death of player%!\" to player")
+@Since("INSERT VERSION")
+public class ExprLastDeath extends SimplePropertyExpression<Player, Date> {
+
+	private final Statistic LAST_DEATH = Statistic.TIME_SINCE_DEATH;
+
+	static {
+		Skript.registerExpression(ExprLastDeath.class, Date.class, ExpressionType.PROPERTY, 
+				"[the] [time of [the]] last death of %players%",
+				"[the] [time of [the]] %players%'[s] last death"
+		);
+		register(ExprLastDeath.class, Date.class, "[the] last death", "players");
+	}
+
+	@SuppressWarnings({"null", "unchecked"})
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		setExpr((Expression<Player>) exprs[0]);
+		return true;
+	}
+
+	@Nullable
+	@Override
+	public Date convert(Player player) {
+		Date date = new Date();
+		date.subtract(Timespan.fromTicks_i(player.getStatistic(LAST_DEATH)));
+		return date;
+	}
+
+	@Nullable
+	@Override
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		if (mode == ChangeMode.DELETE || mode == ChangeMode.REMOVE_ALL)
+			return null;
+		if (mode == ChangeMode.SET)
+			return CollectionUtils.array(Date.class);
+		return CollectionUtils.array(Timespan.class);
+	}
+
+	@SuppressWarnings("null")
+	@Override
+	public void change(Event e, @Nullable Object[] delta, ChangeMode mode) {
+		if (delta == null && mode != ChangeMode.RESET)
+			return;
+		Date now = new Date();
+		for (Player player : getExpr().getArray(e)) {
+			if (mode == ChangeMode.ADD) {
+
+				Date deathTime = convert(player);
+				long add = ((Timespan) delta[0]).getTicks_i();
+				deathTime.add(Timespan.fromTicks_i(add));
+				player.setStatistic(LAST_DEATH, (int) deathTime.difference(now).getTicks_i());
```

Comment:
Would it be a problem if you convert from long -> int? Overflow issues potentially?

Reply:
Possibly - I’ll mess around with it :)

Reply:
it could be but there's nothing he can really do with it, spigot only supports ints for these statistics so that's what we're stuck with. it won't overflow however maybe we can just not do anything if the long > Integer.MAX_VALUE